Tests/GRPCInterceptorsTests/TracingTestsUtilities.swift

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import Tracing
+
+class TestTracer: Tracer {
+  typealias Span = TestSpan
+
+  private var latestTestSpan: TestSpan?
```

Comment:
Hmm, isn't this used unsafely across different threads? I.e. in the tests we read out `latestSpanEvents` but in the interceptors we start spans which creates this.

Reply:
Yes, but my understanding is all tests inside an `XCTestCase` are run sequentially (i.e. no tests inside the same `XCTestCase` will be run in parallel), so this should be safe, as we only start a span once per test. I know this may be a bit of an abuse, but I thought it made it unnecessary to make it thread safe by using a lock.

Reply:
I'm not sure it's circumstantial - the order in which they're run [is explained in the docs](https://developer.apple.com/documentation/xctest/xctestcase/). However I'm okay with making this thread-safe, just pushed the changes.

Reply:
I meant in the sense that we've lined up a set of conditions in which it is safe but they could easily change, e.g.:
- we could run multiple RPCs concurrently in one test
- XCTest could change how it runs tests
- we could migrate to swift-testing which may have different guarantees